tests/integration/mixer/white_black_listing_test.go

```diff
@@ -0,0 +1,108 @@
+// Copyright 2019 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package mixer
+
+import (
+	"testing"
+	"time"
+
+	"fmt"
+	"net/http"
+	"os"
+
+	"fortio.org/fortio/fhttp"
+	"fortio.org/fortio/periodic"
+
+	"istio.io/istio/pkg/test/framework"
+	"istio.io/istio/pkg/test/framework/components/bookinfo"
+	"istio.io/istio/pkg/test/framework/components/environment"
+	"istio.io/istio/pkg/test/framework/components/galley"
+	"istio.io/istio/pkg/test/framework/components/ingress"
+	"istio.io/istio/pkg/test/framework/components/mixer"
+	"istio.io/istio/pkg/test/framework/components/namespace"
+)
+
+func TestWhiteListing(t *testing.T) {
+	framework.Run(t, func(ctx framework.TestContext) {
+		ctx.RequireOrSkip(t, environment.Kube)
+
+		bookinfoNs, err := namespace.New(ctx, "istio-bookinfo", true)
+		if err != nil {
+			t.Fatalf("Could not create istio-bookinfo Namespace; err:%v", err)
+		}
+		d := bookinfo.DeployOrFail(t, ctx, bookinfo.Config{Namespace: bookinfoNs, Cfg: bookinfo.BookInfo})
+		g := galley.NewOrFail(t, ctx, galley.Config{})
+		_ = mixer.NewOrFail(t, ctx, mixer.Config{Galley: g})
+
+		g.ApplyConfigOrFail(
+			t,
+			d.Namespace(),
+			bookinfo.NetworkingBookinfoGateway.LoadOrFailGatewayFileWithNamespace(t, bookinfoNs.Name()))
+		g.ApplyConfigOrFail(
+			t,
+			d.Namespace(),
+			bookinfo.GetDestinationRuleConfigFile(t, ctx).LoadOrFailWithNamespace(t, bookinfoNs.Name()),
+			bookinfo.NetworkingVirtualServiceAllV1.LoadOrFailWithNamespace(t, bookinfoNs.Name()),
+		)
+
+		// Wait for rules to sync.
+		time.Sleep(time.Second * 30)
+
+		ing := ingress.NewOrFail(t, ctx, ingress.Config{Istio: ist})
+
+		res := SendTraffic1(ing, t, "Sending traffic...", "", 2)
+		// Verify you can access productpage right now.
+		if float64(res.RetCodes[http.StatusOK]) == 0 {
+			t.Fatalf("could not accsess productpage")
+		}
+
+		g.ApplyConfigOrFail(
+			t,
+			d.Namespace(),
+			bookinfo.PolicyDenyIPRule.LoadOrFailWithNamespace(t, bookinfoNs.Name()))
+		// Wait for rules to sync.
+		time.Sleep(time.Second * 30)
+
+		// Verify you can't access productpage now.
+		res = SendTraffic1(ing, t, "Sending traffic...", "", 30)
+		if float64(res.RetCodes[http.StatusForbidden]) == 0 {
+			t.Fatalf("didn't get PERMISSION_DENIED error after applying denials rule.")
+		}
+	})
+}
+
+func SendTraffic1(ingress ingress.Instance, t *testing.T, msg, url string, calls int64) *fhttp.HTTPRunnerResults {
```

Comment:
Why `SendTraffic1`?  This appears to be the same method, line by line as `SendTraffic`.

In general, this feels a bit like code smell to me.  Should we update `SendTraffic` to accommodate whatever extra parameterization is needed instead?  Or call this something else?

Reply:
you caught it before I got a chance to rebase :)
I had same functions in 2 reviews and if by chance both of them got merged I kept the name different by 1.. Fixed the code..